lib/script.js

```diff
@@ -28,23 +40,70 @@ function ShowCurrentTime() {
-    .then(result => weather_data.push(result));
+    .then(result => weather_data.unshift(result));
 }
 
 function displayCurrentWeather () {
-    var temperature = Math.ceil(weather_data[0].data.attributes.right_now.apparentTemperature);
-    document.getElementById("current-weather__temperature").innerHTML = temperature + `&deg;`
+    var temperature = Math.round(weather_data[0].data.attributes.right_now.apparentTemperature);
+    document.getElementById("current-weather__temperature").innerHTML = temperature + `&deg;`;
 
     var condition = weather_data[0].data.attributes.right_now.summary;
-    document.getElementById("current-weather__condition").innerHTML = condition
+    document.getElementById("current-weather__condition").innerHTML = condition;
 
     var weather_icon = weather_data[0].data.attributes.right_now.icon;
-    document.getElementById("current-weather__icon").src = `./lib/img/${weather_icon}.png`
-    //  import the icons here
+    document.getElementById("current-weather__icon").src = `./lib/img/${weather_icon}.png`; //import the relevant weather icon
+
+    var current_location = weather_data[0].data.attributes.location;
+    var current_city = current_location.slice(0, -3).toUpperCase();
+    var current_state = current_location.substr(-2).toUpperCase();
+    document.getElementById("current-weather__location").innerHTML = current_city;
+
+    var windspeed = Math.round(weather_data[0].data.attributes.right_now.windSpeed);
+    document.getElementById("current-weather__windspeed").innerHTML = 'Winds at ' + windspeed + ' mph';
+}
+
+
+// when someone clicks a location on the nav, close the nav and populate the new weather data on the cards.
+function listenForCityChange() {
+    document.addEventListener('click', function (event) {
+        // event bubbling with 1 class for multiple elements.
+        if ( event.target.classList.contains('navigation__link')) {
+            document.querySelector('#navi-toggle').click(); // close the nav radial button
+            document.querySelector('.lds-roller').style.visibility = 'visible'; // show spinner again
+            var new_location = event.target.innerHTML; // location = "denver, co"
+            
+            if (new_location === "my location") {
+
+                getWeather(my_city, my_state);
+                pauseThenLoadWeather();
+            } else {
+                var new_city = new_location.slice(0, -4); // "denver"
+                var new_state = new_location.substr(-2); // "co"
+                
+                getWeather(new_city, new_state);
+                pauseThenLoadWeather();
+            }
+            
+        }
+    });
+}
+
+// prompt user to enter their location in the form.
+    // once their info is submitted, save their location to the global object.
+    // load their weather.
+
+function respondToInput() {
+    my_location += document.getElementById('my-location-field').value.toLowerCase(); //sanitize user input to lowercase
```

Comment:
If you are using the google geolocation api, the city/state data does not need to be sanitized. :)

Reply:
oh yeah haha